dbms/src/Storages/Transaction/KVStore.cpp

```diff
@@ -327,6 +327,57 @@ void KVStore::persistRegion(const Region & region, const RegionTaskLock & region
     LOG_FMT_DEBUG(log, "Persist {} done", region.toString(false));
 }
 
+bool KVStore::canFlushRegionData(UInt64 region_id, UInt8 flush_if_possible, bool try_until_succeed, TMTContext & tmt)
```

Comment:
Agree with your options, the names are bits of misleading. Maybe we can change some names.
Since, if there are two functions, other developers may tend to use
```
if isRegionFlushable() {
    checkAndFlushRegionData();
}
```
If there is only one function, developers may not tend to write code like
```
if canFlushData(false) {
    canFlushData(true);
}
```

Reply:
Got it and I think we can avoid the misusage by improve the name further and introduce some syntax like `[[nodiscard]]`.
But my only point is that the semantics of `canFlushRegionData` is a little weird. Because if a function name starts with something like `can`, it's natural to think that it will just do some tests and doesn't change anything.